Fixing #181

kuk0 opened the pull request.

bos commented:
Please fix up this commit stack so that it doesn't have all the whitespace changes in it. I can't review this as it stands.

kuk0 commented:
@bos: dropped whitespace changes and rebased

btw, why "can't review"? all the changes were in separate commits and easy to review separately; i'll assume you meant: "i disagree with the ws changes" in which case, well... meh...

bos commented:
Thanks! Yeah, I wasn't going to read through all the spurious whitespace to find the changes you really made.
